internal/search/index.go

```
@@ -282,15 +301,51 @@ var IndexSchemas = []IndexSchema{
+const schemaFingerprintKey = "pm:indexer:schema:fingerprint"
+
+// SchemaFingerprint is a stable hash of IndexSchemas. The indexer stamps it on
+// every successful Rebuild and compares it at boot; a mismatch means the schema
+// changed (a field added or promoted to SORTABLE/TAG) and the indexes must be
+// dropped+recreated to apply it — FT.CREATE is a no-op on an existing index.
+// Self-maintaining: editing IndexSchemas changes the fingerprint, so there is no
+// version constant to forget to bump (ponytail: the schema IS the version).
+func SchemaFingerprint() string {
+	h := sha256.New()
+	for _, ix := range IndexSchemas {
+		fmt.Fprintf(h, "%s\x00%s\x00", ix.Name, ix.Prefix)
```

golangci-lint (2.12.2)

[error] Error return value of `fmt.Fprintf` is not checked

(errcheck)